pkg/resourceinterpreter/default/thirdparty/resourcecustomizations/batch.volcano.sh/v1alpha1/Job/customizations.yaml

```
@@ -0,0 +1,290 @@
+apiVersion: config.karmada.io/v1alpha1
+kind: ResourceInterpreterCustomization
+metadata:
+  name: declarative-configuration-job
+spec:
+  target:
+    apiVersion: batch.volcano.sh/v1alpha1
+    kind: Job
+  customizations:
+    healthInterpretation:
+      luaScript: >
+        function InterpretHealth(observedObj)
+          if observedObj.status == nil or observedObj.status.state == nil then
+            return false
+          end
+          local phase = observedObj.status.state.phase
+          if phase == nil or phase == '' then
+            return false
+          end
+          if phase == 'Running' or phase == 'Completed' or phase == "Pending" or
+          phase == "Aborting" or phase == "Aborted" or phase == "Restarting" or
+          phase == "Completing" or phase == "Terminating" or phase == "Terminated" then
+            return true
+          end
+          return false
+        end
+    componentResource:
+      luaScript: |
+        local kube = require("kube")
+        local function get(obj, path)
+          local cur = obj
+          for i = 1, #path do
+            if cur == nil then return nil end
+            cur = cur[path[i]]
+          end
+          return cur
+        end
+
+        local function to_num(v, default)
+          if v == nil or v == '' then
+            return default
+          end
+          local n = tonumber(v)
+          if n ~= nil then return n end
+          return default
+        end
+
+        function GetComponents(observedObj)
+          local components = {}
+
+          local tasks = get(observedObj, {"spec","tasks"})
+          if tasks == nil then
+            return components
+          end
+
+          for _, task in ipairs(tasks) do
+            local replicas = to_num(task.minAvailable, 1)
+            requires = kube.accuratePodRequirements(task.template)
+
+            table.insert(components, {
+              name = task.name,
+              replicas = replicas,
+              replicaRequirements = requires
+            })
+          end
```

```suggestion
          for i, task in ipairs(tasks) do
            local replicas = to_num(task.minAvailable, 1)
            local requires = kube.accuratePodRequirements(task.template)

            -- Determine a stable component name with sensible defaults
            local taskName = task.name
            if taskName == nil or taskName == '' then
              -- Use zero-based index for uniqueness: task-0, task-1, ...
              taskName = "task-" .. (i - 1)
            end

            table.insert(components, {
              name = taskName,
              replicas = replicas,
              replicaRequirements = requires
            })
          end
```
The task name in Volcano Job is optional, but Karmada requires each component to have a unique name.